test/Npgsql.Tests/Replication/SafeReplicationTestBase.cs

```diff
@@ -58,7 +58,9 @@ private protected static async Task AssertReplicationCancellation<T>(IAsyncEnume
                 Assert.That(e, Is.AssignableTo<OperationCanceledException>()
                     .With.InnerException.InstanceOf<PostgresException>()
                     .And.InnerException.Property(nameof(PostgresException.SqlState))
-                    .EqualTo(PostgresErrorCodes.QueryCanceled));
+                    .EqualTo(PostgresErrorCodes.QueryCanceled)
+                    .Or.AssignableTo<OperationCanceledException>()
+                    .With.InnerException.InstanceOf<TimeoutException>());
```

Comment:
This is only because older versions don't support PG cancellation for replication, right? If so, it might be better to put this in condition based on the PG version - allowing hard cancellation in modern PG may be a bad idea.

Reply:
Yeah. I was too lazy to pull the connection down there but you're probably right.